capsules/extra/src/ieee802154/framer.rs

```diff
@@ -567,39 +580,84 @@ impl<'a, M: Mac<'a>, A: AES128CCM<'a>> Framer<'a, M, A> {
     /// Advances the reception pipeline if it can be advanced.
     fn step_receive_state(&self) {
         self.rx_state.take().map(|state| {
-            let (next_state, buf) = match state {
-                RxState::Idle => (RxState::Idle, None),
+            let next_state = match state {
+                RxState::Idle => RxState::Idle,
                 RxState::ReadyToDecrypt(info, buf) => {
                     match info.security_params {
                         None => {
                             // `ReadyToDecrypt` should only be entered when
                             // `security_params` is not `None`.
-                            (RxState::Idle, Some(buf))
+                            RxState::Idle
                         }
                         Some((level, key, nonce)) => {
                             let (m_off, m_len) = info.ccm_encrypt_ranges();
                             let (a_off, m_off) = (radio::PSDU_OFFSET, radio::PSDU_OFFSET + m_off);
 
+                            // Crypto setup failed; fail receiving packet and return to idle
                             if self.aes_ccm.set_key(&key) != Ok(())
                                 || self.aes_ccm.set_nonce(&nonce) != Ok(())
                             {
-                                (RxState::Idle, Some(buf))
-                            } else {
-                                let res = self.aes_ccm.crypt(
-                                    buf,
-                                    a_off,
-                                    m_off,
-                                    m_len,
-                                    info.mic_len,
-                                    level.encryption_needed(),
-                                    true,
+                                // No error is returned for the receive function because recv occurs implicitly
```

Comment:
Does this case need to return the buffer to the radio driver? i.e., `self.mac.set_receive_buffer(???);`

Reply:
Yes it should. Good catch!